iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp

```diff
@@ -316,56 +319,72 @@ class AdjustDepthwiseFilterShape : public OpRewritePattern<mhlo::ConvOp> {
 // into
 //
 // %z = "BinaryElementwiseOpT"(%x, %y) : (%[[SHAPE_BEFORE_BCAST]], %[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_BEFORE_BCAST]]
-// %result = "mhlo.broadcast_in_dim"(%z) {} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
+// %result = "mhlo.broadcast_in_dim"(%z) {broadcast_dimensions = %[[BCAST_DIMS]]} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
 //
 // clang-format on
-template <typename BinaryElementwiseOpT>
-class ReorderBroadcastInDimOpAndBinaryElementwiseOp
-    : public OpRewritePattern<BinaryElementwiseOpT> {
+template <typename ElementwiseOpT>
+class ReorderBroadcastInDimOpAndElementwiseOp
+    : public OpRewritePattern<ElementwiseOpT> {
  public:
-  using OpRewritePattern<BinaryElementwiseOpT>::OpRewritePattern;
+  using OpRewritePattern<ElementwiseOpT>::OpRewritePattern;
 
-  LogicalResult matchAndRewrite(BinaryElementwiseOpT op,
+  LogicalResult matchAndRewrite(ElementwiseOpT op,
                                 PatternRewriter &rewriter) const override {
-    auto lhsType = op.lhs().getType().template dyn_cast<ShapedType>();
-    auto rhsType = op.rhs().getType().template dyn_cast<ShapedType>();
-    if (!lhsType || !rhsType || !lhsType.hasStaticShape() ||
-        lhsType != rhsType) {
+    Operation *operation = op.getOperation();
+    if (operation->getNumOperands() < 1 || operation->getNumResults() != 1 ||
```

Comment:
I think we should assert the number of operands is not empty because this is really wrong to add such a pattern to the pass.